src/event_emitter.rs

```diff
@@ -0,0 +1,10 @@
+use tree_sitter_grep::{tree_sitter::Node, SupportedLanguage};
+
+pub type Event = String;
+
+pub trait EventEmitter<'a> {
+    fn name(&self) -> String;
```

Comment:
I think this is going to change, started thinking that there will be an event emitter factory trait and that's where the event name and a list of the events that it emits would be exposed

And then we can see which event emitters people are actually using and only instantiate those from the factory